functors task (2)/student/include/impl.h

```diff
@@ -53,31 +53,39 @@ class DataBrowser
             > m_dataReaders;
 
     /**
-     * @todo SafeCall idiom
-     * You should implement this method, that is responsible for request validation.
-     * If request is valid (user exists and a valid selector exists),
-     * this method calls passed callback with a valid selector pointer
+     * Checks user id for existing and valid selector and performs the call of data request.
+     * Passed @param f must have the signature bool(const std::unique_ptr<IDataSelector>&)
      */
     template <typename T>
-    bool safeCall(const std::string &userId, T&& f) const
-    {
-        static_assert(std::is_same<decltype(f(nullptr)), bool>::value, "Provided Callable must return bool");
-        // find reader
-        // check for errors
-        // call functor
-        return false;
-    }
+    bool safeCall(const std::string &userId, T&& f) const;
 
     /**
-     * @todo You should implement this method, that is responsible for a "general" data requesting
-     * General data request will return bool, accept the selector and output param
-     * You must adapt functors before invocation
+     * General data requesting via a single interface.
```

Comment:
Is it a suggestion for the task description improvement? If it is, please create pull request

Reply:
No, it's just something like documentation after I have implemented the functional. It helps me remember technical details. It is not always clear, for example, which signature is appropriate for "template" functionals params.
As a variant, I would rather set the type of functional param a bit explicitly, for example, 
```
template<typename Output>
bool invokeDataRequest(
  std::function<bool(IDataSelector*, Output)> method,
  const std::unique_ptr<IDataSelector>& selector,
  Output& result) const
```
but for study purposes that is nice as is :) really challenging task.